api_tests/users/views/test_user_message_institutional_access.py

```diff
@@ -151,5 +153,95 @@ def test_admin_cannot_message_user_outside_institution(
         """
         res = app.post_json_api(url_without_affiliation, payload, auth=institutional_admin.auth, expect_errors=True)
         assert res.status_code == 409
-        assert 'Can not send to recipient that is not affiliated with the provided institution.'\
-               in res.json['errors'][0]['detail']['user']
+        assert ('Cannot send to a recipient that is not affiliated with the provided institution.'
+                in res.json['errors'][0]['detail']['user'])
+
+    @mock.patch('osf.models.user_message.send_mail')
+    def test_cc_institutional_admin(
+            self,
+            mock_send_mail,
+            app,
+            institutional_admin,
+            institution,
+            url_with_affiliation,
+            user_with_affiliation,
+            payload
+    ):
+        """
+        Ensure CC option works as expected, sending messages to all institutional admins except the sender.
+        """
+        mock_send_mail.return_value = mock.MagicMock()
+
+        # Enable CC in the payload
+        payload['data']['attributes']['bcc_sender'] = True
+
+        # Perform the API request
+        res = app.post_json_api(
+            url_with_affiliation,
+            payload,
+            auth=institutional_admin.auth,
+        )
+        assert res.status_code == 201
+        user_message = UserMessage.objects.get()
+
+        assert user_message.is_sender_BCCed
+        # Two emails are sent during the CC but this is how the mock works `send_email` is called once.
+        mock_send_mail.assert_called_once_with(
+            to_addr=user_with_affiliation.username,
+            cc_addr=[institutional_admin.username],
```

Comment:
This may be handled later on, but in the case of a bcc, both the sender and the recipient should be bcc'd, otherwise the sender will be able to see the recipient's email address.